patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch

```diff
@@ -257,22 +186,185 @@
+ 
+ #if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+ void ChromePasswordManagerClient::CheckProtectedPasswordEntry(
+@@ -568,24 +569,11 @@ void ChromePasswordManagerClient::CheckP
+     const std::string& username,
+     const std::vector<std::string>& matching_domains,
+     bool password_field_exists) {
+-  safe_browsing::PasswordProtectionService* pps =
+-      GetPasswordProtectionService();
+-  if (!pps)
+-    return;
+-
+-  pps->MaybeStartProtectedPasswordEntryRequest(
+-      web_contents(), web_contents()->GetLastCommittedURL(), username,
+-      password_type, matching_domains, password_field_exists);
+ }
+ #endif  // defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+ 
+ #if defined(SYNC_PASSWORD_REUSE_WARNING_ENABLED)
+ void ChromePasswordManagerClient::LogPasswordReuseDetectedEvent() {
+-  safe_browsing::PasswordProtectionService* pps =
+-      GetPasswordProtectionService();
+-  if (pps) {
+-    pps->MaybeLogPasswordReuseDetectedEvent(web_contents());
+-  }
+ }
+ #endif  // defined(SYNC_PASSWORD_REUSE_WARNING_ENABLED)
+ 
+@@ -692,7 +680,9 @@ void ChromePasswordManagerClient::OnInpu
+   if (key_event.windows_key_code == (ui::VKEY_V & 0x1f)) {
+     OnPaste();
+   } else {
++#if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+     password_reuse_detection_manager_.OnKeyPressed(key_event.text);
++#endif  // defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+   }
+ #endif  // defined(OS_ANDROID)
+ }
+@@ -806,9 +796,11 @@ void ChromePasswordManagerClient::OnPast
+   base::string16 text;
+   clipboard->ReadText(ui::ClipboardBuffer::kCopyPaste, &text);
+   was_on_paste_called_ = true;
++#if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+   password_reuse_detection_manager_.OnPaste(std::move(text));
++#endif  // defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+ }
+-#endif
++#endif  // !defined(OS_ANDROID)
+ 
+ gfx::RectF ChromePasswordManagerClient::GetBoundsInScreenSpace(
+     const gfx::RectF& bounds) {
+--- a/chrome/browser/password_manager/chrome_password_manager_client.h
++++ b/chrome/browser/password_manager/chrome_password_manager_client.h
+@@ -181,7 +181,7 @@ class ChromePasswordManagerClient
+   void OnImeTextCommittedEvent(const base::string16& text_str) override;
+ #endif  // defined(OS_ANDROID)
+ 
+-#if defined(ON_FOCUS_PING_ENABLED)
++#if defined(ON_FOCUS_PING_ENABLED) && BUILDFLAG(FULL_SAFE_BROWSING)
+   void CheckSafeBrowsingReputation(const GURL& form_action,
+                                    const GURL& frame_url) override;
+   safe_browsing::PasswordProtectionService* GetPasswordProtectionService()
+--- a/chrome/browser/resource_coordinator/tab_ranker/tab_score_predictor.cc
++++ b/chrome/browser/resource_coordinator/tab_ranker/tab_score_predictor.cc
+@@ -310,17 +310,6 @@ void TabScorePredictor::LazyInitialize()
+     DCHECK_EQ(preprocessor_config_->feature_indices().size(),
+               static_cast<std::size_t>(tfnative_model::FEATURES_SIZE));
+   }
+-
+-  if (type_ == kPairwiseScorer) {
```

Comment:
We also need to revert this change as part of adding back the binary file exception.